### DISCUSSION TITLE
Meeting notes #2, 2022-09-20

### DISCUSSION THREAD

**Discussion by jongfeel** [wisdom08/dev-community#18] [General]
### 미진한 작업 진행률에 대한 문제점

- 소셜 로그인 기능을 구현 후에 게시판 기능을 구현해야 하는데, 소셜 로그인 기능 에서 로그인 유지 상태를 잘 모르고 있고 문제가 되고 있다. 그래서 소셜 로그인 기능 구현을 일반 로그인 기능으로 대체해서 빠르게 구현하고 소셜 로그인 기능을 나중에 구현하려고 한다.
  - 사실 소셜 로그인 기능을 몰라서 게시판 기능을 구현 못하는 게 문제가 되지 않음
  - 소셜 로그인 기능에서 무엇이 문제인지 내용을 공유하고 이를 해결하려는 생각을 많이 안한 게 문제

### 테스트 코드 작성에 대한 어려움

- 테스트 코드 작성하는 강의도 듣고 했는데 실제 테스트 코드를 짤 수는 없었음
  - 테스트 코드 작성하는 방법을 몰라서 테스트코드를 못짜고 있는게 아니라 무엇을 테스트해야 하는지 생각을 못하고 있을 가능성이 높음
  - 테스트 코드 자체는 어처구니 없을 정도로 당연한 기능에 대해 당연한 결과가 나오는지 확인하는 코드를 작성하는 것이고, 이 테스트 수십개가 모여서 소프트웨어의 안정성을 받쳐 주는거라 생각하는게 좋음
  - 어떤 기능에 대해서 데이터와 원하는 결과를 생각하고 그걸 코드로 구현해 보는 연습을 해보는 것이 필요

### 개선해서 해볼 것들

- 하루 2시간, 2주 동안 할 수 있는 작업의 범위를 정하고, 날짜별로 진행할 작업 내용을 정리해 보자. 뽀모도로 단위로 계획.
- 테스트 코드 작성, 그 전에 내가 만드려는 기능에 무엇을 테스트 할 것인지를 적어 보고 진행
- 작은 단위의 issue 내용을 만들고 그 코드를 pull request로 작성
  - 반대로 코드를 작성하고 무슨 작업들이었는지 적어 보는 것도 좋음 => 단, 코드 내용을 설명하는 걸로 적지 말자
- 생각의 변화와 생각지 못한 내용, 추가해야 하는 작업 등을 issue 코멘트에 잘 적어서 관리

**Comment by wisdom08** [comment]
코드를 짜기 전에 미리 작전을 잘 세우고 그 작전에 대한 기록을 잘 해볼게요! 
감사합니다.

**Reply by jongfeel** [reply]
작전(설계)을 세우고 진행하는 것도 중요하지만
진행 중에 변경 사항에 대한 업데이트 기록을 잘 남기는 것도 중요합니다

**Reply by wisdom08** [reply]
넵 그 부분도 개선해보겠습니다🫡
